_posts/2023-06-06-Client-cert-auth-without-roles.adoc

```diff
@@ -0,0 +1,252 @@
+For this step, let us create a filesystem realm called `idsRealm`. We will enter into this realm our identity with 
+name `client-cert-user` which we will map the certificate into. 
+
+```
+/subsystem=elytron/filesystem-realm=idsRealm:add(path=ids, relative-to=jboss.server.config.dir)
+/subsystem=elytron/filesystem-realm=idsRealm:add-identity(identity=client-cert-user)
+```
+
+Next, we need to configure our server-ssl-context. This is a collection of ciphers, TLS settings and certificates, like the 
+ones we will be using to authenticate our server and the client. Since the `applicationSSC` server-SSL-context already exists, 
+we will not be adding one. Instead, we will be editing its properties using the `write-attribute` command: 
+
+```
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=key-manager, value=twoWayKM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=trust-manager, value=twoWayTM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=need-client-auth, value=true)
+```
+
+Now, when we are establishing the connection, elytron will know where to go to look for the keystore and the truststore. 
+Setting `want-client-auth` to true indicates that a client needs to be authenticated in order to establish a connection. 
+
+Next, we will need to configure a decoder that will be able to decode the CA Certification. We will be using 
+a regex-principal-transformer chained with the x500 attribute principal decoder to map usernames with the given pattern 
+into our client-cert-user. 
+
+```
+/subsystem=elytron/x500-attribute-principal-decoder=cnDecoder:add(attribute-name=CN, maximum-segments=1)
+/subsystem=elytron/regex-principal-transformer=myRegexTransformer:add(pattern="[a-zA-Z]+\\.[a-zA-Z]+\\.[0-9]+", replacement="client-cert-user")
+/subsystem=elytron/chained-principal-transformer=myChainedTransformer:add(principal-transformers=[cnDecoder, myRegexTransformer])
+```
+
+Now we will use a security realm to use the filesystem realm fom before: 
+```
+/subsystem=elytron/security-domain=clientCertDomain:add(realms=[{realm=idsRealm}], default-realm=idsRealm, pre-realm-principal-transformer=myChainedTransformer, permission-mapper=default-permission-mapper)
+```
+
+Afterwards, use the following commands to configure the CLIENT_CERT HTTP mechanism on elytron:
+```
+/subsystem=elytron/configurable-http-server-mechanism-factory=configuredCert:add(http-server-mechanism-factory=global, properties={org.wildfly.security.http.skip-certificate-verification=true})
+/subsystem=elytron/http-authentication-factory=clientCertAuth:add(http-server-mechanism-factory=configuredCert, security-domain=clientCertDomain , mechanism-configurations=[{mechanism-name=CLIENT_CERT}])
+```
+
+And finally, we need to configure the undertow subsystem to be able to establish a connection. Please run the following commands
```

Comment:
s/to be able to establish a connection/to make use of our `http-authentication-factory`